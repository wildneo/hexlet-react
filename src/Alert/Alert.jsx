import React from 'react';
import cn from 'classnames';

export default class Alert extends React.Component {
  render() {
    const { type, text } = this.props;
    const alertClass = cn('alert', `alert-${type}`);

    return (
      <div className={alertClass} role="alert">
        {text}
      </div>
    );
  }
}
