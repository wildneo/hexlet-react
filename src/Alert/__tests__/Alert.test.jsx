import React from 'react';
import renderer from 'react-test-renderer';
import Alert from '../Alert';

test('Alert 1', () => {
  const params = {
    type: 'danger',
    text: 'text 1',
  };
  const component = renderer.create(<Alert {...params} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Alert 2', () => {
  const params = {
    type: 'dark',
    text: 'text 2',
  };
  const component = renderer.create(<Alert {...params} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
