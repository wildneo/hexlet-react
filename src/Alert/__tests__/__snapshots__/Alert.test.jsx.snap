// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert 1 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  text 1
</div>
`;

exports[`Alert 2 1`] = `
<div
  className="alert alert-dark"
  role="alert"
>
  text 2
</div>
`;
